api/infrastructure/parallelcluster-api.yaml

```diff
@@ -969,47 +1064,70 @@ Resources:
         ZipFile: |
           import cfnresponse
           import boto3
+          import random
+          import string
 
-          # Suggestion is to instantiate clients in global context to reuse them across Lambda invocations
           ecr = boto3.client('ecr')
-
-          def get_tag_from_image_arn(image_arn):
-              """ Returns the Docker image tag given an ImageBuilder image ARN
-
-              Arn format: arn:aws:imagebuilder:<region>:<account id>:image/.../<version>/<build number>
-              Tag format: <version>-<build number>
-              """
-              parts = image_arn.split('/')
-              version = parts[-2]
-              build_number = parts[-1]
-              return '{}-{}'.format(version, build_number)
+          imagebuilder = boto3.client('imagebuilder')
+
+          def get_image_ids(repository_name, version):
+              image_ids = []
+              paginator = ecr.get_paginator('list_images')
+              response_iterator = paginator.paginate(repositoryName=repository_name, filter={'tagStatus': 'TAGGED'})
+              for response in response_iterator:
+                  image_ids.extend([{'imageTag': image_id['imageTag']} for image_id in response['imageIds'] if image_id['imageTag'].startswith(version)])
```

Comment:
to make this more robust we need to probably check for startswith(version + "-") otherwise this check fails with versions such as 3.0.1 that are included in 3.0.10